packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -240,4 +255,176 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
       LIMIT 1;
     `);
   }
+
+  async getUsersWhoHaveMadeDialogues(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+      WITH all_dialogue_authors AS
+        (SELECT (UNNESTED->>'userId') AS _id
+            FROM "Posts" p, UNNEST("coauthorStatuses") unnested
+            WHERE p."collabEditorDialogue" IS TRUE 
+            AND p."draft" IS FALSE
+        UNION
+        SELECT P."userId" as _id
+            FROM "Posts" as P
+            WHERE P."collabEditorDialogue" IS TRUE
+            AND p."draft" IS FALSE
+        )
+      SELECT U.*
+      FROM "Users" AS U
+      INNER JOIN all_dialogue_authors ON all_dialogue_authors._id = U._id
+    `)
+  }
+
+  async getUsersWhoHaveOptedInToDialogueFacilitation(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+        SELECT *
+        FROM "Users" u
+        WHERE u."optedInToDialogueFacilitation" IS TRUE
+    `)
+  }  
+
+  async getUsersTopUpvotedUsers(userId:string): Promise<UpvotedUser[]> {
+    return this.getRawDb().any(
+      `
+      WITH "CombinedVotes" AS (
+        -- Joining Users with Posts and Votes
+        SELECT
+            public."Users"._id,
+            public."Users".username,
+            public."Users"."displayName",
+            public."Votes".power,
+            CASE
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'bigDownvote' THEN -6
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'smallDownvote' THEN -2
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'neutral' THEN 0
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'smallUpvote' THEN 2
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'bigUpvote' THEN 6
```

Comment:
What's up with the weighting on these?

Reply:
Leaving for Jacob

Reply:
They're not in the db. Discussed solution with oli then implemented as follows